File: test-codebase/all_issues.py
#!/usr/bin/env python3
"""
All Issues Combined - A file with all types of problems
This is what you'd run AI Patch Doctor on to see all checks
"""

import os
import time
import openai
from openai import OpenAI

client = OpenAI(api_key=os.getenv('OPENAI_API_KEY'))

def problematic_api_call():
    """An API call with multiple issues."""
    
    # STREAMING ISSUES:
    # - No timeout
    # - Missing streaming headers
    
    # RETRY ISSUES:
    # - No exponential backoff
    # - Too many retries
    
    # COST ISSUES:
    # - No max_tokens
    # - Large prompt
    
    # TRACEABILITY ISSUES:
    # - No request ID
    # - No idempotency key
    
    max_retries = 10  # Too many
    
    for attempt in range(max_retries):
        try:
            # Large prompt (cost issue)
            large_prompt = "Write a detailed analysis of " + "AI " * 500
            
            # Streaming with issues
            stream = client.chat.completions.create(
                model="gpt-4",  # Expensive
                messages=[{"role": "user", "content": large_prompt}],
                stream=True,
                # MISSING: max_tokens
                # MISSING: timeout
                # MISSING: headers for streaming
                # MISSING: idempotency-key
            )
            
            for chunk in stream:
                if chunk.choices[0].delta.content:
                    print(chunk.choices[0].delta.content, end='', flush=True)
            
            print("\n✅ Success!")
            return
            
        except openai.RateLimitError:
            # Bad retry logic
            wait = 1  # Should be exponential
            print(f"Rate limited, waiting {wait}s...")
            time.sleep(wait)
            
        except Exception as e:
            print(f"Error: {e}")
            if attempt < max_retries - 1:
                time.sleep(1)
            else:
                raise

if __name__ == "__main__":
    print("⚠️  This code has multiple issues!")
    print("Run: pipx run ai-patch-doctor doctor")
    print("Or: npx ai-patch doctor")
    print("\nSelect 'all' or specific checks to see issues detected.")

File: test-codebase/all_issues.js
#!/usr/bin/env node
/**
 * All Issues Combined - A file with all types of problems
 * This is what you'd run AI Patch Doctor on to see all checks
 */

const OpenAI = require('openai');

const client = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY
});

async function problematicApiCall() {
  // STREAMING ISSUES:
  // - No timeout
  // - Missing streaming headers
  
  // RETRY ISSUES:
  // - No exponential backoff
  // - Too many retries
  
  // COST ISSUES:
  // - No max_tokens
  // - Large prompt
  
  // TRACEABILITY ISSUES:
  // - No request ID
  // - No idempotency key
  
  const maxRetries = 10;  // Too many
  
  for (let attempt = 0; attempt < maxRetries; attempt++) {
    try {
      // Large prompt (cost issue)
      const largePrompt = 'Write a detailed analysis of ' + 'AI '.repeat(500);
      
      // Streaming with issues
      const stream = await client.chat.completions.create({
        model: 'gpt-4',  // Expensive
        messages: [{ role: 'user', content: largePrompt }],
        stream: true,
        // MISSING: max_tokens
        // MISSING: timeout
        // MISSING: headers for streaming
        // MISSING: idempotency-key
      });
      
      for await (const chunk of stream) {
        if (chunk.choices[0]?.delta?.content) {
          process.stdout.write(chunk.choices[0].delta.content);
        }
      }
      
      console.log('\n✅ Success!');
      return;
      
    } catch (error) {
      if (error.status === 429) {
        // Bad retry logic
        const wait = 1;  // Should be exponential
        console.log(`Rate limited, waiting ${wait}s...`);
        await new Promise(resolve => setTimeout(resolve, wait * 1000));
      } else {
        console.log(`Error: ${error.message}`);
        if (attempt < maxRetries - 1) {
          await new Promise(resolve => setTimeout(resolve, 1000));
        } else {
          throw error;
        }
      }
    }
  }
}

if (require.main === module) {
  console.log('⚠️  This code has multiple issues!');
  console.log('Run: pipx run ai-patch-doctor doctor');
  console.log('Or: npx ai-patch doctor');
  console.log('\nSelect \'all\' or specific checks to see issues detected.');
}
